labelbox/data/serialization/coco/image.py

```diff
@@ -26,15 +33,44 @@ def get_image_id(label: Label, idx: int) -> int:
     return idx
 
 
-def get_image(label: Label, image_root: Path, image_id: str) -> CocoImage:
-    path = Path(image_root, f"{image_id}.jpg")
-    if not path.exists():
-        im = Image.fromarray(label.data.value)
-        im.save(path)
-        w, h = im.size
-    else:
-        w, h = imagesize.get(str(path))
-    return CocoImage(id=image_id, width=w, height=h, file_name=Path(path.name))
+def get_image(
+    label: Label,
+    image_path: Path,
+    image_id: str,
+    cloud_provider=None,
+    azure_storage_container=None,
+) -> CocoImage:
+    """
+    Extracts dimensions from an image provided by 'image_path'. The path may be
+    local or a path in a cloud storage service. In that case,
+    'image_path' should be used together with 'cloud_provider'.
+    Currently, possible values for 'cloud_provider' are:
+        - azure
+    """
+
+    if not cloud_provider:
+        path = Path(image_path, f"{image_id}.jpg")
+        if not path.exists():
+            im = Image.fromarray(label.data.value)
+            im.save(path)
+            w, h = im.size
+        else:
+            w, h = imagesize.get(str(path))
+        return CocoImage(
+            id=image_id, width=w, height=h, file_name=Path(label.data.url).name
+        )
+
+    elif cloud_provider == "azure":
+        conn = get_connection_string()
+        client = create_blobstorage_client(conn, azure_storage_container)
+        file_path = extract_file_path(label.data.url, azure_storage_container)
+        logger.info(f"Downloading {file_path} from blobstorage")
+        image = client.download_blob(file_path).readall()
+        h = Image.open(io.BytesIO(image)).height
+        w = Image.open(io.BytesIO(image)).width
```

Comment:
Why create an `Image` twice?

```suggestion
        img = Image.open(io.BytesIO(image))
        h, w = img.height, img.width
```